### ISSUE TITLE
Spell Checker: Problems with hyphen, comments, urls and commands/arguments

### ISSUE THREAD

**Issue by Stimmenhotel** [texstudio-org/texstudio#1429]
### Environment
- TeXstudio: 3.0.4
- Qt: 5.12.3
- OS: Win10
- TeX distribution: Miktex 20.1x
- No Language Tool running

I am about to finish my thesis and tried to use the spell checker tool of texstudio to find any obvious mistakes i made.
But unfortunately it was showing much much more mistakes than the syntax highlight.
This behavior could be wanted, but let me describe it with a few samples.

### Expected behavior
- Ignoring comments, ignoring arguments inside [ ] and { } if they are green. (Section/paragraph arguments should be checked).
- Same behavior between error-syntax-highlighting and spell-checker-tool.

### Actual behavior
- **Hyphen** are not used correctly (seems only to be a syntax highlight problem, as the spell checker won't show it anymore once added to my custom dictionary)
![grafik](https://user-images.githubusercontent.com/15821524/104193668-a8a4ee80-5420-11eb-8652-f5215125717d.png) ![grafik](https://user-images.githubusercontent.com/15821524/104193702-afcbfc80-5420-11eb-8112-7b3a5992294a.png)
- Any **comment**, even cryptical ones like **URLs** are reported
![grafik](https://user-images.githubusercontent.com/15821524/104193364-52d04680-5420-11eb-9fe8-2ae1ef8bec78.png) ![grafik](https://user-images.githubusercontent.com/15821524/104193410-5e237200-5420-11eb-92f6-108611b7bef3.png)
- **Commands**, or more accurate **arguments**, are reported
![grafik](https://user-images.githubusercontent.com/15821524/104193268-36340e80-5420-11eb-8483-0e1b9bd6adc7.png) ![grafik](https://user-images.githubusercontent.com/15821524/104193305-41873a00-5420-11eb-9279-a4532bc117e6.png) ![grafik](https://user-images.githubusercontent.com/15821524/104193496-785d5000-5420-11eb-998e-9e24552ef2e8.png)

### How to reproduce

Adding these lines in a file:
```
%https://uic.org/rail-system/gsm-r/
\begin{itemize}[label=\small{\textcolor{gray}{$ \blacksquare $}}]
	\item \textit{\acrfull{ac:gsmr}}
	\item \textit{\acrfull{ac:etcs}}
	\item \textit{\acrfull{ac:etml}}
\end{itemize}
\paragraph{GSM-R}
"`Class-B-Systeme"'
```

and try the right-click correction and the spell-checker tool on it

**Comment by Stimmenhotel** [comment]
Just found another one.

If using German quotation marks  **"'** these are counted into the wrong word (at least the "):
![grafik](https://user-images.githubusercontent.com/15821524/104195550-0e927580-5423-11eb-8393-b321a6b5336c.png)

Added the left one into the dict:
![grafik](https://user-images.githubusercontent.com/15821524/104195586-1baf6480-5423-11eb-848c-e082da93e38e.png)

**Comment by sunderme** [comment]
fixed with 57b9fb33ed11b7559ab9c1347e630dc88b958f40
